mobile/native/src/ln_dlc/channel_status.rs

```diff
@@ -104,100 +93,63 @@ async fn channel_status(node: impl Borrow<Node>) -> Result<ConcreteChannelStatus
+        })
+        | Some(SignedChannel {
+            state: SignedChannelState::SettledOffered { .. },
             ..
-        } => {
-            match subchannels
-                .iter()
-                .find(|subchannel| subchannel.channel_id == channel_id)
-            {
-                // We never had a subchannel associated with the latest LN channel.
-                None => ConcreteChannelStatus::NotOpen,
-                Some(subchannel) => match SubChannelState::from(subchannel) {
-                    // The subchannel was never fully open.
-                    SubChannelState::Rejected | SubChannelState::Opening => {
-                        ConcreteChannelStatus::NotOpen
-                    }
-                    // The subchannel was closed offchain.
-                    SubChannelState::CollabClosed => ConcreteChannelStatus::NotOpen,
-                    // The subchannel was close on-chain.
-                    SubChannelState::ForceClosed => ConcreteChannelStatus::NotOpen,
-                    // The subchannel is somehow still open even though the LN channel is not.
-                    SubChannelState::Open | SubChannelState::CollabClosing => {
-                        ConcreteChannelStatus::Inconsistent
-                    }
-                    // The subchannel is still open being closed on-chain.
-                    SubChannelState::ForceClosing => ConcreteChannelStatus::LnDlcForceClosing,
-                },
-            }
-        }
-        // If the LN channel does not have an ID associated with it, we assume that it's still not
-        // open, for it should not be usable yet.
-        Channel {
-            channel_id: None, ..
-        } => ConcreteChannelStatus::NotOpen,
-        Channel {
-            channel_state: ChannelState::Announced,
+        })
+        | Some(SignedChannel {
+            state: SignedChannelState::SettledReceived { .. },
+            ..
+        })
+        | Some(SignedChannel {
+            state: SignedChannelState::SettledAccepted { .. },
+            ..
+        })
+        | Some(SignedChannel {
+            state: SignedChannelState::RenewOffered { .. },
+            ..
+        })
+        | Some(SignedChannel {
+            state: SignedChannelState::RenewAccepted { .. },
+            ..
+        })
+        | Some(SignedChannel {
+            state: SignedChannelState::RenewConfirmed { .. },
+            ..
+        }) => ConcreteChannelStatus::Pending,
+        Some(SignedChannel {
+            state: SignedChannelState::Closing { .. },
+            ..
+        })
+        | Some(SignedChannel {
+            state: SignedChannelState::CollaborativeCloseOffered { .. },
+            ..
+        }) => ConcreteChannelStatus::Closing,
+        Some(SignedChannel {
+            state: SignedChannelState::Established { .. },
+            ..
+        })
+        | Some(SignedChannel {
+            state: SignedChannelState::RenewFinalized { .. },
```

Comment:
🤔 Is this really how it works? Doesn't the channel go back to `Established` after this one? I'd consider this state part of `Pending`.

Reply:
Afaik we end in `RenewFinalized` or `Settled` after any off-chain action and `Established` is only once the channel has been set up in the first place

Reply:
Hmm, we should verify this.